src/variable_info.hpp

```diff
@@ -16,98 +16,119 @@
 
 #pragma once
 
-/** Information on a WML variable. */
-#include <string>
 #include "config.hpp"
 #include "variable_info_detail.hpp"
 
-class invalid_variablename_exception : public std::exception
+#include <string>
+#include <type_traits>
+
+// Restrict alias to local scope.
+namespace
 {
-public:
-	invalid_variablename_exception() : std::exception() {}
-	const char* what() const NOEXCEPT
-	{
-		return "invalid_variablename_exception";
-	}
-};
+namespace impl = variable_info_implementation;
```

Comment:
I'm thinking what you really want is for this namespace alias to be local to the class somehow. Not entirely sure if that's possible though.